### DISCUSSION TITLE
From shared hosting to Docker container

### DISCUSSION THREAD

**Discussion by mikae1** [FreshRSS/FreshRSS#5044] [Q&A] [answered]
Hi, I’ve used FreshRSS for many years. I’ve hosted it on Dreamhost shared by just uploading the dir and running it from the public dir. I used a SQLite db.

Now I have an RPi4 with Docker, Docker Compose and Portainer installed on my local network. I’d like to migrate the dir from Dreamhost shared to a Docker container. 

Does anybody have a recommendation on how to do this?

**Comment by Alkarex** [comment, marked as answer]
For a Raspberry Pi, regarding the hardware, I recommend putting your FreshRSS data (i.e. your Docker volumes) on an external SSD as it will be much faster and reliable than the SD card.

To install your new instance, see our [Docker read-me](https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md).

When you are done installing your new FreshRSS instance:

1) you can import your old data by:
  a) either copy your user's old SQLite `./FreshRSS/data/users/you/db.sqlite` file over your user's new SQLite file;
  b) or use our [`cli/import-sqlite-for-user.php`](https://github.com/FreshRSS/FreshRSS/blob/edge/cli/README.md#user) command line (works for whichever database you have on your new system: SQLite, PostgreSQL, MariaDB, MySQL).

2) You can also import your old user configuration by copying `./FreshRSS/data/users/you/config.php` over your user's new configuration file.

For that, you need to access old files from inside your new Docker instance. This can be done in multiple ways:

a) Using the [`docker cp`](https://docs.docker.com/engine/reference/commandline/cp/) command
b) Use a local folder as a volume instead of a proper Docker volume (this is also an easy way to put your data on an external SSD)
c) A few other variants...

**Reply by mikae1** [reply]
I'm very thankful for your reply! I will give it a try during the weekend and report back. 

My only worry now is that the version of FreshRSS that ran on my shared hosting is older than the most recent Docker version and that this might cause some kind of mismatch. We'll soon see!

**Reply by mikae1** [reply]
Works! Thanks again! Just did 1.a) and 2.b).

What would be the process for setting up automated feed updating when using Docker?

I’m using the Argon One SSD case BTW. My SQLite db is >600 MB and it seems to be running great. Only  using microSD for backups. 8 GB RPi4.

**Reply by Alkarex** [reply]
There is a built-in cron available in our official Docker images, which can be enabled by a `CRON_MIN` environment variable. Check our documentation: https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md
We also have a few more advanced cron options https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#cron-job-to-automatically-refresh-feeds

**Reply by mikae1** [reply]
> There is a built-in cron available in our official Docker images, which can be enabled by a `CRON_MIN` environment variable.

Thanks! Forgot to answer to this. The refresh that's happening without me changing anything seems to be a sane default, so have not touched it. Still _very_ happy with this setup. 🙂
